docs/sources/flow/reference/components/loki.source.heroku.md

```diff
@@ -0,0 +1,136 @@
+---
+aliases:
+- /docs/agent/latest/flow/reference/components/loki.source.heroku
+title: loki.source.heroku
+---
+
+# loki.source.heroku
+
+`loki.source.heroku` listens for Heroku messages over TCP connections
+and forwards them to other `loki.*` components.
+
+The component starts a new heroku listener for the given `listener`
+block and fans out incoming entries to the list of receivers in `forward_to`.
+
+Before using `loki.source.heroku`, Heroku should be configured with the URL where the Agent will be listening. Follow the steps in [Heroku HTTPS Drain docs](https://devcenter.heroku.com/articles/log-drains#https-drains) for using the Heroku CLI with a command like the following:
+
+```
+heroku drains:add [http|https]://HOSTNAME:PORT/heroku/api/v1/drain -a HEROKU_APP_NAME
+```
+
+Multiple `loki.source.heroku` components can be specified by giving them
+different labels.
+
+## Usage
+
+```river
+loki.source.heroku "LABEL" {
+  listener {
+    address = "LISTEN_ADDRESS"
+    port = PORT
+  }
+  ...
+
+  forward_to = RECEIVER_LIST
+}
+```
+
+## Arguments
+
+`loki.source.heroku` supports the following arguments:
+
+Name                     | Type                   | Description          | Default | Required
+------------------------ | ---------------------- | -------------------- | ------- | --------
+`use_incoming_timestamp` | `bool`                 | Whether or not to use the timestamp received from Heroku. | `false` | no
+`labels`                 | `map(string)`          | The labels to associate with each received Heroku record. | `{}`    | no
+`forward_to`             | `list(LogsReceiver)`   | List of receivers to send log entries to.                 |         | yes
+`relabel_rules`          | `RelabelRules`         | Relabeling rules to apply on log entries.                 | `{}`    | no
+
+The `relabel_rules` field can make use of the `rules` export value from a
+`loki.relabel` component to apply one or more relabeling rules to log entries
+before they're forwarded to the list of receivers in `forward_to`.
+
+## Blocks
+
+The following blocks are supported inside the definition of `loki.source.heroku`:
+
+Hierarchy | Name | Description | Required
+--------- | ---- | ----------- | --------
+listener | [listener] | Configures a listener for Heroku messages. | yes
+
+[listener]: #listener-block
+
+### listener block
+
+The `listener` block defines the listen address and port where the listener
+expects Heroku messages to be sent to.
+
+Name                     | Type          | Description | Default | Required
+------------------------ | ------------- | ----------- | ------- | --------
+`address`                | `string`      | The `<host>` address to listen to for heroku messages. | `0.0.0.0` | no
+`port`                   | `int`         | The `<port>` to listen to for heroku messages. | | yes
+
+Example Block
+
+```river
+listener {
+  address  = "127.0.0.1"
+  port = 8080
+}
+```
```

Comment:
Nitpick, but we don't usually give out examples for inner blocks, and this is a pretty simple one, so not sure if we should keep it.